p2p/host/autonat/autonat_test.go

```diff
@@ -209,46 +208,43 @@ func TestAutoNATObservationRecording(t *testing.T) {
 		t.Fatalf("failed to subscribe to event EvtLocalRoutabilityPublic, err=%s", err)
 	}
 
-	addr, _ := ma.NewMultiaddr("/ip4/127.0.0.1/udp/1234")
-	an.recordObservation(autoNATResult{network.ReachabilityPublic, addr})
+	an.recordObservation(network.ReachabilityPublic)
 	if an.Status() != network.ReachabilityPublic {
 		t.Fatalf("failed to transition to public.")
 	}
 
 	expectEvent(t, s, network.ReachabilityPublic, 3*time.Second)
 
 	// a single recording should have confidence still at 0, and transition to private quickly.
-	an.recordObservation(autoNATResult{network.ReachabilityPrivate, nil})
+	an.recordObservation(network.ReachabilityPrivate)
 	if an.Status() != network.ReachabilityPrivate {
 		t.Fatalf("failed to transition to private.")
 	}
 
 	expectEvent(t, s, network.ReachabilityPrivate, 3*time.Second)
 
 	// stronger public confidence should be harder to undo.
-	an.recordObservation(autoNATResult{network.ReachabilityPublic, addr})
-	an.recordObservation(autoNATResult{network.ReachabilityPublic, addr})
+	an.recordObservation(network.ReachabilityPublic)
+	an.recordObservation(network.ReachabilityPublic)
 	if an.Status() != network.ReachabilityPublic {
 		t.Fatalf("failed to transition to public.")
 	}
-
 	expectEvent(t, s, network.ReachabilityPublic, 3*time.Second)
 
-	an.recordObservation(autoNATResult{network.ReachabilityPrivate, nil})
+	an.recordObservation(network.ReachabilityPrivate)
 	if an.Status() != network.ReachabilityPublic {
 		t.Fatalf("too-extreme private transition.")
 	}
 
-	// don't emit events on observed address change
-	newAddr, _ := ma.NewMultiaddr("/ip4/127.0.0.1/udp/12345")
-	an.recordObservation(autoNATResult{network.ReachabilityPublic, newAddr})
+	// Don't emit events if reachability hasn't changed
+	an.recordObservation(network.ReachabilityPublic)
 	if an.Status() != network.ReachabilityPublic {
 		t.Fatalf("reachability should stay public")
 	}
 	select {
 	case <-s.Out():
 		t.Fatal("received event without state transition")
-	case <-time.After(300 * time.Millisecond):
+	case <-time.After(1 * time.Second):
```

Comment:
Do we really need to increase the timeout here?

Reply:
I'll change this back to 300ms. I haven't seen this failing so far.